### PR TITLE
4740/add asymmetric key encryption to registration and login

### PR DESCRIPTION
**Pull request by EVogel1999**
This PR adds the encryption and decryption module to user service.  This creates a new route GET /keys which gets a public key to encrypt with.  For login and registration routes, if you send up a body that has the schema:
```
{ publicKey: string, data: string }
```
It will automatically try to decrypt given the public string, otherwise it will return whatever body was given.  Completes story [4740/add-asymmetric-key-encryption-to-registration-and-login](https://app.clubhouse.io/clarkcan/story/4740/add-asymmetric-key-encryption-to-registration-and-login).

### CODE REVIEW

**Review by jmulholland4014 — APPROVED** [review]

File: src/drivers/MongoDriver.ts [hunk 1]
```diff
@@ -449,7 +450,6 @@ export default class MongoDriver implements DataStore {
       const arr = await organizations.toArray();
       return arr;
     } catch (e) {
-      console.log(e);
```
Comment (pzalep1) [inline src/drivers/MongoDriver.ts:452]: Thank you.

File: src/drivers/RouteHandler.ts [hunk 2]
```diff
@@ -223,50 +173,49 @@ export default class RouteHandler {
...
-                this.dataStore,
+                RouteHandler.datastore,
                 responder
               );
               break;
             default:
               responder.sendOperationError('Invalid action');
           }
         } catch (e) {
-          console.log(e);
```
Comment (pzalep1) [inline src/drivers/RouteHandler.ts:254]: Thank you.

File: src/encryption-module/datastore/EncryptionDatastore.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        return EncryptionDatastore.instance;
+    }
+    
+    /**
+     * Saves a public and private key pair in the database
+     * 
+     * @param pair The pair to save
+     */
+    async saveKeyPair(pair: KeyPair): Promise<void> {
+        await this.client.collection(COLLECTIONS.KEY_PAIRS).insertOne({
```
Comment (pzalep1) [inline src/encryption-module/datastore/EncryptionDatastore.ts:33]: Is this new collection in onion?

**Review by mfranz1 — APPROVED** [review]

File: src/encryption-module/datastore/EncryptionDatastore.ts (hunk 3 above)
Comment (EVogel1999) [reply src/encryption-module/datastore/EncryptionDatastore.ts:33]: Yes it is.

**Review by jouypo23 — APPROVED** [review]

**Review by pzalep1 — APPROVED** [review]

**Pull request merged**